File: 66-PlusOne/66-PlusOne.java
// Last updated: 16/10/2025, 04:30:25
class Solution {
    public int[] plusOne(int[] digits) {
        int n = digits.length;
        int ans[] = new int[n];
        int carry = 1;
        for(int i = n - 1; i >= 0; --i){
            ans[i] = (digits[i] + carry) % 10;
            carry = (digits[i] + carry) / 10;
        }
        if(carry != 0){
            ans = Arrays.copyOf(ans, ans.length + 1);
            ans[ans.length - 1] = 0;
            ans[0] = 1;
        }
        return ans;
    }
}

File: 66-PlusOne/66-PlusOne.cpp
// Last updated: 16/10/2025, 04:30:30
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        vector<int> v;
        for(int i:digits)v.push_back(i);
        reverse(v.begin(),v.end());
        int car=1;
        for(int i=0;i<v.size();i++)
        {
            v[i]+=car;
            car=v[i]/10;
            v[i]%=10;
        }
        if(car)v.push_back(car);
        reverse(v.begin(),v.end());
        return v;
    }
};